rfcs/0132-meson-builds-nix.md

```diff
@@ -0,0 +1,216 @@
+2. Also by design, the implementation language is never exposed to the meson DSL. Because of it, the reimplementation of Meson in other programming languages becomes way more facilitated.
+   1. Indeed, Muon is an alternative implementation of Meson written in C.
+3. As part of this evaluation of this bootstrap route, we should also evaluate the current bootstrap route, in order to have a fair comparison.
+
+Further, in principle the same criticisms and answers can be laid out for Ninja; however, Ninja is written in C++, a language already used to implement Nix. Therefore the bootstrap route suffers little to no alteration here.
+
+## Learning the new system
+
+A somewhat subjective but non-negligible issue is the entry barrier of this new build system. Switching from a known build system to one unknown is not without its problems.
+
+However, the Meson development team strives to keep the DSL easy to learn and pleasurable to use. It should not be hard to become familiar with the Python-esque syntax of meson, and its functional, stateless approach is certainly a feature highly appreciated by the Nix community as a whole.
+
+The huge advantages of implementing Meson surpass the small drawbacks of learning it.
+
+## Source code changes
+
+Further to the obvious inclusion of meson files (and the removal of the old quasi-autotools ones), there is a reasonable expectation of code refactoring.
+
+However, such refactorings are completely validated on the long term goals of Nix, in particular the improvements on portability.
+
+## End users
+
+The most known end user of Nix is certainly Nixpkgs. However, there are many other Linux distributions that already keep Nix on their repositories (15 families, according to Repology). There is also a reasonable expectation of affecting those package managers' devteams.
+
+However, most (if not all) of those distributions already have Meson and its companion tool Ninja in their respective package databases (53 families, according to Repology), given that many open source projects use them as build system.
+
+## Transition between old and new build infrastructure
+
+The transition between between the old and new build systems should be as smooth and controlled as possible.
+
+# Alternatives
+[alternatives]: #alternatives
+
+The alternatives are
+
+- Do nothing
+
+  It would keep the current code confusing and harder to work with, as stated on the motivation section.
+
+- Use CMake
+
+  Indeed, CMake has many noteworthy advantages:
+
+  - Supports typical high level idiomatic constructions.
+  - Can generate GMake- and Ninja-compatible scripts.
+    - By design, Meson does not provide a Make backend.
+  - Both Meson and CMake support Microsoft Windows NT and Apple MacOS platforms
+    - As well as MSVC and XCode programming environments.
+
+  - On the other hand, CMake DSL cmake language is arguably more complex and cumbersome.
```

Comment:
```suggestion
  - On the other hand, the CMake DSL language is considerably more complex and cumbersome than meson.
```

Reply:
This is deliberately open to interpretation. Many people said modern cmake is more readable than before.

Reply:
More readable than before != actually readable. :D

Anyway, there are problems modern cmake cannot ever solve, such as the "do you need to quote this or not" question. Also, I just love reading lists as "element1;element2;element3", don't you? (Don't forget: whether this is a list or a single string depends on where it's used, not how it's defined.)